cl/search/signals.py

```diff
@@ -656,3 +662,47 @@ def update_court_cache(sender, instance: Court, created: bool, **kwargs):
 
     if created:
         logger.error("Create a courthouse for new court '%s'", instance.id)
+
+
+def clean_docket_number_raw_and_update_redis_cache(
+    docket: Docket,
+):
+    r = get_redis_interface("CACHE")
+
+    docket_number, docket_id_llm = clean_docket_number_raw(
+        docket_id=docket.id,
+        docket_number_raw=docket.docket_number_raw,
+        court_id=docket.court_id,
+    )
+    docket.docket_number = docket_number
+    docket.save(update_fields=["docket_number"])
+
+    # Add to redis cache for later processing
+    if docket_id_llm:
+        # Use a Redis lock to avoid race conditions when getting and updating the llm_batch.
+        set_key = "docket_number_cleaning:llm_batch"
+        lock_key = f"{set_key}:lock"
+        lock_value = acquire_redis_lock(r, lock_key, 60 * 1000)
+        try:
+            r.sadd(set_key, docket_id_llm)
+        finally:
+            release_redis_lock(r, lock_key, lock_value)
+
+
+@receiver(
+    post_save,
+    sender=Docket,
+    dispatch_uid="handle_docket_number_raw_cleaning",
+)
+def handle_docket_number_raw_cleaning(
+    sender, instance: Docket, created=False, update_fields=None, **kwargs
+):
+    if not settings.DOCKET_NUMBER_CLEANING_ENABLED:
+        # Only perform cleaning if enabled
+        return
+
+    # Only clean if the docket was was non-recap source and newly created or docket_number_raw has changed
+    changed = not created and "docket_number_raw" in update_fields
+    non_recap_sources = instance.source in instance.DOCKET_NUM_CLEAN_SOURCES()
```

Comment:
I reviewed the behavior of `DOCKET_NUM_CLEAN_SOURCES` and noticed that it lists all available sources except for `RECAP`. Is that correct?

If so, I think we don’t need `DOCKET_NUM_CLEAN_SOURCES` and can instead do something like:
`non_recap_sources = instance.source != Docket.RECAP`

If the approach should be different and exclude all sources that involve any combination of the RECAP source, we can use the existing `NON_RECAP_SOURCES`.